tests/export_genesis.sh

```diff
@@ -53,31 +53,57 @@ gaiad_upgrade () {
     su gaia -c "echo \"abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon abandon art\" | ~/.gaia/cosmovisor/current/bin/gaiad --output json keys add val --keyring-backend test --recover 2> ~/.gaia/validator.json" # Use stderr until gaiad use stdout
     
     # Test to see if gaia is building blocks
+    echo "Testing block productions on version: $f_gaia_version"
     su gaia -c "tests/test_block_production.sh 127.0.0.1 26657 $(($f_initial_height+10))"
     if [ $? -ne 0 ]
     then
         echo "Failed to build blocks on version: $f_gaia_version"
-        f_build_block=0
+        f_pass=0
     else
         echo "Test building blocks for version: $f_gaia_version"
-        f_build_block=1
+        f_pass=1
     fi
 
-    if [ $f_build_block -eq 1 ]
+    # Test upgrading
+    if [ $f_pass -eq 1 ]
     then
         echo "Testing upgrade"
         su gaia -c "tests/test_software_upgrade.sh 127.0.0.1 26657 $f_upgrade_version"
         if [ $? -ne 0 ]
         then
             f_message="Upgrade failed from $f_gaia_version to $f_upgrade_version"
+            f_pass=0
+        else
+            f_message="Upgrade Successful from $f_gaia_version to $f_upgrade_version"
+            f_pass=1
+        fi
+    else
+        echo "SKIPPING testing upgrade due to failed job"
+    fi
+
+    # Happy path - transaction testing after upgrade
+    echo "Testing happy path on version: $f_upgrade_version"
+    if [ $f_pass -eq 1 ]
+    then
+        cp tests/test_tx_stateful.sh ~gaia/
```

Comment:
Won't `~gaia/` generate an incorrect path? Like if the current user is called `user`, the path would be `/home/usergaia/`. Is that desired? From the line below I would assume the correct statement would be manually specifying `/home/gaia/`:

```suggestion
        cp tests/test_tx_stateful.sh /home/gaia/
```

Reply:
Whoops, I see that now. Thanks for explaining.